src/librustdoc/json/conversions.rs

```diff
@@ -19,32 +19,35 @@ impl JsonRenderer<'_> {
         let item_type = ItemType::from(&item);
         let deprecation = item.deprecation(self.tcx);
         let clean::Item { source, name, attrs, kind, visibility, def_id } = item;
-        match *kind {
-            clean::StrippedItem(_) => None,
-            kind => Some(Item {
-                id: def_id.into(),
-                crate_id: def_id.krate.as_u32(),
-                name: name.map(|sym| sym.to_string()),
-                source: self.convert_span(source),
-                visibility: self.convert_visibility(visibility),
-                docs: attrs.collapsed_doc_value().unwrap_or_default(),
-                links: attrs
-                    .links
-                    .into_iter()
-                    .filter_map(|clean::ItemLink { link, did, .. }| {
-                        did.map(|did| (link, did.into()))
-                    })
-                    .collect(),
-                attrs: attrs
-                    .other_attrs
-                    .iter()
-                    .map(rustc_ast_pretty::pprust::attribute_to_string)
-                    .collect(),
-                deprecation: deprecation.map(Into::into),
-                kind: item_type.into(),
-                inner: kind.into(),
-            }),
-        }
+        let inner = match *kind {
+            clean::ItemKind::ExternCrateItem(a) => ItemEnum::ExternCrateItem {
+                name: name.as_ref().unwrap().to_string(),
+                rename: a.map(|x| x.to_string()),
+            },
+            clean::StrippedItem(_) => return None,
+            x => x.into(),
```

Comment:
Why not make the ExternCrateItem part of the `Into` impl?

Reply:
Because we need the item's name, which I removed in this same PR.